src/native/libs/System.Native/pal_networking.c

```diff
@@ -1439,8 +1439,9 @@ int32_t SystemNative_Send(intptr_t socket, void* buffer, int32_t bufferLen, int3
 
     ssize_t res;
 #if defined(__APPLE__) && __APPLE__
+    int maxProtoRetry = 4;
```

Comment:
A comment in the source as to why this exists and how 4 was chosen would be helpful.